core/web/keys_controller.go

```diff
@@ -82,6 +100,15 @@ func (kc *keysController[K, R]) Delete(c *gin.Context) {
 		jsonAPIError(c, http.StatusInternalServerError, err)
 		return
 	}
+
+	// Emit audit log, determine if Terra or Solana key
+	switch any(key).(type) {
+	case terrakey.Key:
+		kc.lggr.Auditf(logger.TERRA_KEY_DELETED, map[string]interface{}{"terraID": keyID})
+	case solkey.Key:
+		kc.lggr.Auditf(logger.SOLANA_KEY_DELETED, map[string]interface{}{"solanaID": keyID})
+	}
```

Comment:
Hmmm....we'll need to find another way to achieve these without the type switches. Both to ensure that it remains extensible (can be used without being modified), and so that it does not have to directly import any of the concrete types, which could lead to cycles in the short term and would prevent decoupling modules in the long term.

One simplification would be to use a common audit key for all types, and limit the chain specific stuff to the _data_. Could we start with that? Or is it necessary for the audit keys to be different for each chain type?

Reply:
Hmm, that's a good point for the import, but we would like to have distinct audit logs for each event. We could conflate them potentially here because conveniently the terra public keys have a prefix - do you think this is a blocker here?

Reply:
Well short-term we should at least be injecting these differences through the constructors, and then just using it here without any type switch. If we keep the _keys_ unique, could we consider simplifying the data format to be more general? `"id": keyID`

Reply:
Ah okay - I agree, it looks like we do get context for Solana vs Terra that is stored in the struct from `resourceName` in `func NewKeysController`, but I think it's better to use the URL path here instead. We could also potentially reuse `getFieldNameForKey` in `core/services/keystore/master.go`? 

A bigger issue is that without the casting I don't have access to the public key, I looked into added `PublicKey()` to the Key interface but it's not straightforward to implement for all the key types. `ocrkey` for example has two public key fields that I'm not sure how to cast. What do you think?

Also good point for consolidating to "id" - updated 🙂